cores/esp32/esp32-hal-uart.c

```diff
@@ -107,6 +125,14 @@ void uartEnableInterrupt(uart_t* uart)
     uart->dev->int_ena.rxfifo_tout = 1;
     uart->dev->int_clr.val = 0xffffffff;
 
+    if(arg != NULL){
+        (*arg) = malloc(sizeof(uart_interrupt_t));
+        (*arg)->func = func;
+        (*arg)->dev = uart;
+        (*arg)->user_arg = user_arg;
+        _uart_interrupt_array[uart->num] = (*arg);
```

Comment:
what if `_uart_interrupt_array[uart->num]` already exists? Leak?

Reply:
uartDisableInterrupt get's called before the uartEnableInterrupt function get's called, so the `_uart_interrupt_array[uart->num]` should be deleted.
Just for safety I'll create a if statement that deletes the value if it is still present